src/cliques.c

```diff
@@ -0,0 +1,628 @@
+              0L);
+  }
+  if (!IS_LIST(include_obj) && include_obj != Fail) {
+    ErrorQuit("the 5th argument <include> must be a list or fail, not %s,",
+              (Int) TNAM_OBJ(include_obj),
+              0L);
+  } else if (IS_LIST(include_obj)) {
+    for (Int i = 1; i <= LEN_LIST(include_obj); ++i) {
+      if (!ISB_LIST(include_obj, i)) {
+        ErrorQuit("the 5th argument <include> must be a dense list,", 0L, 0L);
+      } else if (!IS_POS_INT(ELM_LIST(include_obj, i))) {
+        ErrorQuit("the 5th argument <include> must only contain positive "
+                  "integers, but found %s in position %d,",
+                  (Int) TNAM_OBJ(ELM_LIST(include_obj, i)),
+                  i);
+      } else if (INT_INTOBJ(ELM_LIST(include_obj, i)) 
+                 > DigraphNrVertices(digraph_obj)) {
+        ErrorQuit("in the 5th argument <include> position %d is out of range, "
+                  "must be in the range [1, %d],",
+                  i,
+                  DigraphNrVertices(digraph_obj));
+      } else if (INT_INTOBJ(
+                     POS_LIST(include_obj, ELM_LIST(include_obj, i), INTOBJ_INT(0)))
+                 < i) {
+        ErrorQuit("in the 5th argument <include> position %d is a duplicate,", i, 0L);
+      }
+    }
+  }
+  if (!IS_LIST(exclude_obj) && exclude_obj != Fail) {
+    ErrorQuit("the 6th argument <exclude> must be a list or fail, not %s,",
+              (Int) TNAM_OBJ(exclude_obj),
+              0L);
+  } else if (IS_LIST(exclude_obj)) {
+    for (Int i = 1; i <= LEN_LIST(exclude_obj); ++i) {
+      if (!ISB_LIST(exclude_obj, i)) {
+        ErrorQuit("the 6th argument <exclude> must be a dense list,", 0L, 0L);
+      } else if (!IS_POS_INT(ELM_LIST(exclude_obj, i))) {
+        ErrorQuit("the 6th argument <exclude> must only contain positive "
+                  "integers, but found %s in position %d,",
+                  (Int) TNAM_OBJ(ELM_LIST(exclude_obj, i)),
+                  i);
+      } else if (INT_INTOBJ(ELM_LIST(exclude_obj, i)) 
+                 > DigraphNrVertices(digraph_obj)) {
+        ErrorQuit("in the 6th argument <exclude> position %d is out of range, "
+                  "must be in the range [1, %d],",
+                  i,
+                  DigraphNrVertices(digraph_obj));
+      } else if (INT_INTOBJ(
+                     POS_LIST(exclude_obj, ELM_LIST(exclude_obj, i), INTOBJ_INT(0)))
+                 < i) {
+        ErrorQuit("in the 6th argument <exclude> position %d is a duplicate,", i, 0L);
+      }
+    }
+  }
+  if (max_obj != True && max_obj != False) {
+    ErrorQuit("the 7th argument <max> must true or false, not %s,",
+              (Int) TNAM_OBJ(max_obj),
+              0L);
+  }
+  if (!IS_INTOBJ(size_obj) && size_obj != Fail) {
+    ErrorQuit("the 8th argument <size> must be an integer "
+              "or fail, not %s,",
+              (Int) TNAM_OBJ(size_obj),
+              0L);
+  } else if (IS_INTOBJ(size_obj) && INT_INTOBJ(size_obj) <= 0) {
```

Comment:
Just use `IS_POS_INTOBJ` -- or even better, combined this check with the previous one, and say `must be a positive small integer` in the error message?